### DISCUSSION TITLE
Creating public URL for files in JuiceFS

### DISCUSSION THREAD

**Discussion by arpowers** [juicedata/juicefs#3229] [General]
Hey there, 
Wondering there is a methodology we could use to provide public URLs for large files we store with JFS? 

In our case, we would love to use it for images, and other large files but they all need public URLs that can be provided for the client.

**Comment by davies** [comment]
The WebDAV (plain HTTP with/without basic auth) should work for you, please check `juicefs webdav --help`

**Comment by arpowers** [comment]
@davies THANKS! i'll give it a shot

**Comment by yuhr123** [comment]
Hi @arpowers, there are some security tips for you: 

As WebDAV is a file sharing protocol that includes read, write, and delete permissions, if anonymous sharing is configured, please do not use it on public networks to prevent attackers from accessing and destroying your data through WebDAV. For your requirement to generate Public URLs for files stored on JuiceFS, in addition to the WebDAV protocol included with the software, you can also use web servers such as Apache, Nginx, and Caddy Server to configure the JuiceFS mount point as a publicly accessible and read-only directory.
